File: Leetcode/leetcode-0145-BinaryTreePostorderTraversal-[tree]-[hard]/Solution.recursive.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function (root) {
    let result = [];
    if (root === null) {
        return result;
    }

    postorder(root, result);
    return result;
};

let postorder = (root, result) => {
    if (root === null) {
        return;
    }

    postorder(root.left, result);
    postorder(root.right, result);
    result.push(root.val);
};

File: Leetcode/leetcode-0590-NaryTreePostorderTraversal-[tree]-[easy]/Solution.recursive.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        List<Integer> result = new ArrayList<>();
        if (root == null) return result;
        traverse(root, result);
        return result;
    }
    
    public void traverse(Node root, List<Integer> result) {
        if (root == null) return;
        
        List<Node> curLevel = root.children;
        for (Node node : curLevel) {
            traverse(node, result);
        }
        
        result.add(root.val);
    }
}